openai-model-provider/proxy/proxy.go

```diff
@@ -87,25 +87,31 @@ func Run(cfg *Config) error {
 
 	mux := http.NewServeMux()
 
-	// Register custom path handlers first
-	for path, handler := range cfg.CustomPathHandleFuncs {
-		mux.HandleFunc(path, handler)
-	}
-
 	// Register default handlers only if they are not already registered
 	if _, exists := cfg.CustomPathHandleFuncs["/{$}"]; !exists {
 		mux.HandleFunc("/{$}", s.healthz)
 	}
-	if _, exists := cfg.CustomPathHandleFuncs["/v1/models"]; !exists {
+	if handler, exists := cfg.CustomPathHandleFuncs["/v1/models"]; !exists {
```

Comment:
@iwilltry42 can you review this?

Reply:
LGTM!
I don't like all those if-else statements, but have no cleaner solution right now (unfortunately, mux doesn't take care of the path sorting and neither allows overrides nor looking up what paths are already handled.